test/fixtures/servers/http.js

```diff
@@ -48,4 +48,26 @@ httpServer.on('/custom-reason', function (req, res) {
     res.end();
 });
 
+httpServer.on('/file-upload', function (req, res) {
+    if (req.method === 'POST') {
+        var body = '';
+
+        req.on('data', function (data) {
+            body += data;
```

Comment:
Don't convert Buffer to a string, instead concat buffers and calculate using `byteLength`.

Reply:
This unnecessarily casts the Buffer to String.